src/undo/LineStyleUndoAction.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Xournal++
+ *
+ * Undo action for line style changes (Edit selection)
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <vector>
+
+#include "model/LineStyle.h"
+
+#include "UndoAction.h"
+
+class LineStyleUndoActionEntry;
+class Stroke;
+class Layer;
+class Redrawable;
+
+class LineStyleUndoAction: public UndoAction {
+public:
+    LineStyleUndoAction(const PageRef& page, Layer* layer);
+    virtual ~LineStyleUndoAction();
+
+public:
+    virtual bool undo(Control* control);
+    virtual bool redo(Control* control);
+    virtual string getText();
+
+    void addStroke(Stroke* s, LineStyle originalStyle, LineStyle newStyle);
+
+private:
+    std::vector<LineStyleUndoActionEntry*> data;
```

Comment:
There's no need to hold pointers, just store them directly.

Reply:
```suggestion
    std::vector<LineStyleUndoActionEntry> data;
```